highlight/events-explainer.md

```diff
@@ -89,69 +80,69 @@ All [PointerEvent types](https://www.w3.org/TR/pointerevents2/#pointer-event-typ
 
 ### [Interaction with elements and element pointer events](#interaction-with-elements-and-element-pointer-events)
 
-Whenever an element receives pointer input, if a range within that element is also hit, the range's HighlightRangeGroup should receive a pointer event that corresponds to the pointer event received by the element (except in the case of pointer capture - see section on [interaction with pointer capture](#interaction-with-pointer-capture) below). To determine the set of ranges that are hit, we hit test the boxes for elements as usual and after finding the hit element, we find the ranges that cover that hit.
+Whenever an element receives pointer input, if a range within that element is also hit, the range's Highlight should receive a pointer event that corresponds to the pointer event received by the element (except in the case of pointer capture - see section on [interaction with pointer capture](#interaction-with-pointer-capture) below). To determine the set of ranges that are hit, we hit test the boxes for elements as usual and after finding the hit element, we find the ranges that cover that hit.
 
-Elements and HighlightRangeGroups should receive separate pointer events. The reason is that adding HighlightRangeGroup as a target of the same pointer event that elements receive would involve adding a non-element type to what is currently an element-only event. This could be a breaking change for the web. Additionally, a HighlightRangeGroup callback would be able to prevent the event from reaching an element by calling stopPropogation, which may also break web author expectations on what events an element should receive.
+Elements and Highlights should receive the same pointer events. This means that a Highlight callback would be able to prevent the event from reaching an element by calling stopPropogation, as well as canceling the event via preventDefault.

```

Comment:
To my understanding, this concern has two parts. The first part is about enabling a non-element type to be the target property of an event. This would indeed be a risky change, by having the highlight and range properties in HighlightPointerEvent, the element's target can remain the same.

The second part is about preventing default and stopping propagation. Under this event delivery model, in which the highlighted range is virtually part of the elements' hierarchy, I think it is reasonable to assume that canceling or stopping propagation from the range's listener would affect its behavior or delivery to an element.

Let me know whether my assumptions are correct and if this answers your questions.

Reply:
> by having the highlight and range properties in HighlightPointerEvent, the element's target can remain the same.

This is a nice idea for getting around the potential compatibility issues of a non-element UI pointer event target. What do you think should happen with `currentTarget` in this approach, when running the callback for a listener set on a `Highlight`? Setting `event.currentTarget` to the highlight (instead of the element) doesn't seem like it has the same compat risk as with `event.target`, since only listeners set on `Highlight` will see the non-element `currentTarget`. It's a little strange that for the event handler on the `Highlight`, `event.target` would not equal `event.currentTarget`, even though the `Highlight` is the "real" target of the event. But I'm not sure how much of a problem that is.

On the other hand, if we always set `event.currentTarget` to the element even when the event is dispatched to the `Highlight`, we're breaking the normal behavior where `currentTarget` changes as the event is bubbled to different targets in the DOM.

> I think it is reasonable to assume that canceling or stopping propagation from the range's listener would affect its behavior or delivery to an element.

I agree with this.

I'm ok with switching the main content of the document to this new approach, but if you do so could you mention the old approach in an open question? I think we'd want to discuss the tradeoffs here with the working group.

Reply:
`event.currentTarget` seems like the right alternative for `target` assuming there are no web-breaking issues. Removed `highlight` property in favor of `currentTarget` and added an open question to track the previous approach.